ports/stm32/boards/Passport/modules/data_codecs/compact_seedqr_codec.py

```diff
@@ -0,0 +1,78 @@
+# SPDX-FileCopyrightText: © 2020 Foundation Devices, Inc. <hello@foundationdevices.com>
+# SPDX-License-Identifier: GPL-3.0-or-later
+#
+# compact_seedqr_decoder.py
+#
+# Compact SeedQR codec
+#
+
+from .data_encoder import DataEncoder
+from .data_decoder import DataDecoder
+from .data_sampler import DataSampler
+from .qr_type import QRType
+
+
+class CompactSeedQRDecoder(DataDecoder):
+    def __init__(self):
+        self.data = None
+
+    def add_data(self, data):
+        self.data = data
+
+    def estimated_percent_complete(self):
+        return 1 if self.data is not None else 0
+
+    def is_complete(self):
+        return self.data is not None
+
+    def decode(self, **kwargs):
+        return self.data
+
+    def qr_type(self):
+        return QRType.QR
+
+
+class CompactSeedQREncoder(DataEncoder):
+    def __init__(self, _args):
+        self.data = None
+
+    def encode(self, data, is_binary=False, max_fragment_len=None):
+        import trezorcrypto
+        import math
+        binary_str = ""
+
+        # Get indices in binary string of 11 bits per index
+        for word in data:
+            index = trezorcrypto.bip39.find_word(word)
+            index_bin = '{:011b}'.format(index)
+            # binary_str += '{:011}'bin(index).split('b')[1].zfill(11)
+            binary_str += index_bin
+
+        # Exclude checksum bits
+        if len(data) == 24:
+            binary_str = binary_str[:-8]
+
+        if len(data) == 12:
+            binary_str = binary_str[:-4]
+
+        # Convert to binary
+        as_bytes = bytearray()
```

Comment:
Use a fixed-size array instead since we know the size.  This can avoid multiple allocations that can occur with `append()`.